Angular/src/app/_services/data.service.ts

```diff
@@ -26,6 +27,28 @@ export class DataService {
     });
   }
 
+  deleteRole(roleId: any) {
+    return axios
+      .delete(deleteRoleURL, { 
+        data:{roleId:roleId},
+        headers: this.getHeaders() })
+      .then(function(response) {
+        return response;
+      })
+      .catch(function(error) {
+        if (error.response) {
```

Comment:
Will error response ever be false (null, empty etc)? Think in your catch here, you can simply have:

```
return {
    error: error.response.data.Message,
    statusCode: error.response.statusCode
};
```